server/routes/api.js

```diff
@@ -755,4 +741,117 @@ cron.schedule("0 0 * * * *", async () =>
     fetchEventData();
 });
 
+// Once a day check if there are members whose membership will be outdated in two weeks
+// send them a two week notice. Also check if there are members whose membership will
+// be outdated in a week send them a weeks notice.
+cron.schedule('0 0 0 * * *', async () =>
+{
+    const twoWeeksFromNow = new Date();
+    twoWeeksFromNow.setDate(twoWeeksFromNow.getDate() + 14);
+
+    const weekFromNow = new Date();
+    weekFromNow.setDate(weekFromNow.getDate() + 7);
+
+    // Find all the members whose membership expires in two weeks
+    const members = await Member.find({});
+
+    if (members)
+    {
+        let expiresInTwoWeeks = [];
+        let expiresInOneWeek = [];
+        let expiresToday = [];
+
+        members.forEach((member) =>
+        {
+            if (member.membershipExpirationDate.getDate() == twoWeeksFromNow.getDate())
+            {
+                expiresInTwoWeeks.push(member.email);
+            }
+            else if (member.membershipExpirationDate.getDate() == weekFromNow.getDate())
+            {
+                expiresInOneWeek.push(member.email);
+            }
+            else if (member.membershipExpirationDate.getDate() == new Date().getDate())
+            {
+                expiresToday.push(member.email);
+            }
+        })
+
+        if (expiresToday.length > 0) 
+        {
+            const message = "Your membersship to AssocEase expires today!"
```

Comment:
A small typo in "membersship", should only have one s :D Applies to all messages

Reply:
Fixed that!